src/PIL/JxlImagePlugin.py

```diff
@@ -0,0 +1,180 @@
+from __future__ import annotations
+
+import struct
+from io import BytesIO
+
+from . import Image, ImageFile
+
+try:
+    from . import _jxl
+
+    SUPPORTED = True
+except ImportError:
+    SUPPORTED = False
+
+
+## Future idea:
+## it's not known how many frames does animated image have
+## by default, _jxl_decoder_new will iterate over all frames without decoding them
+## then libjxl decoder is rewinded and we're ready to decode frame by frame
+## if OPEN_COUNTS_FRAMES is False, n_frames will be None until the last frame is decoded
+## it only applies to animated jpeg xl images
+# OPEN_COUNTS_FRAMES = True
+
+
+def _accept(prefix):
+    is_jxl = (
+        prefix[:2] == b"\xff\x0a"
+        or prefix[:12] == b"\x00\x00\x00\x0c\x4a\x58\x4c\x20\x0d\x0a\x87\x0a"
+    )
+    if is_jxl and not SUPPORTED:
+        return "image file could not be identified because JXL support not installed"
+    return is_jxl
+
+
+class JxlImageFile(ImageFile.ImageFile):
+    format = "JPEG XL"
+    format_description = "JPEG XL image"
+    __loaded = 0
+    __logical_frame = 0
+
+    def _open(self):
+        self._decoder = _jxl.PILJxlDecoder(self.fp.read())
+
+        width, height, mode, has_anim, tps_num, tps_denom, n_loops, n_frames = (
+            self._decoder.get_info()
+        )
+        self._size = width, height
+        self.info["loop"] = n_loops
+        self.is_animated = has_anim
+
+        self.n_frames = None
+        self._tps_dur_secs = 1
+        if not self.is_animated:
+            self.n_frames = 1
+        elif n_frames > 0:
+            self.n_frames = n_frames
+            self._tps_dur_secs = tps_num / tps_denom
```

Comment:
Can we flip these, "positive" first?
```suggestion
        if n_frames >= 1:
            self.n_frames = n_frames
            self._tps_dur_secs = tps_num / tps_denom
        elif not self.is_animated:
            self.n_frames = 1
```

Reply:
I'm feeling so inclined into:
```
        self.n_frames = 1
        if self.is_animated and n_frames > 0:
            self.n_frames = n_frames
            self._tps_dur_secs = tps_num / tps_denom
```

Reply:
Maybe the first one with fewer levels of indentation.